File: functions/libgcrypt/variants/go/4_montgomery_reduce_0.go
package main

func montgomeryReduce(a int32) int16 {
    var t int16
    t = int16(a) * -3327
    t = int16((a - int32(t)*3329) >> 16)
    return t
}


func main(){
    montgomeryReduce(0)
}

File: functions/libgcrypt/variants/c/4_montgomery_reduce_0.c
#include <stdint.h>

int16_t montgomery_reduce(int32_t a) {
  int16_t t;

  t = (int16_t)((uint16_t)a * 32209);
  t = (int16_t)((a - t * 3329) >> 16);
  return t;
}


int main(){
    montgomery_reduce(0);
}
